bgpd/bgp_route.c

```diff
@@ -3157,17 +3158,11 @@ int bgp_update(struct peer *peer, struct prefix *p, uint32_t addpath_id,
 
 	bgp_attr_dup(&new_attr, attr);
 
-	/* Apply incoming route-map.
-	 * NB: new_attr may now contain newly allocated values from route-map
-	 * "set"
-	 * commands, so we need bgp_attr_flush in the error paths, until we
-	 * intern
-	 * the attr (which takes over the memory references) */
+	/* Apply incoming route-map. */
 	if (bgp_input_modifier(peer, p, &new_attr, afi, safi, NULL,
 		label, num_labels, rn) == RMAP_DENY) {
 		peer->stat_pfx_filter++;
 		reason = "route-map;";
-		bgp_attr_flush(&new_attr);
```

Comment:
While this change is correct (i.e. the `bgp_attr_flush` will never do anything), I would strongly prefer keeping it here for consistency and avoiding future bugs (e.g. if something else is added above that changes `new_attr`.)